src/main/webapp/app/communication/course-conversations-components/dialogs/conversation-detail-dialog/tabs/conversation-info/conversation-info.component.ts

```diff
@@ -221,6 +238,49 @@ export class ConversationInfoComponent implements OnInit, OnDestroy {
             });
     }
 
+    onMuteToggle(): void {
+        const currentMuted = this.activeConversation()?.isMuted ?? false;
+        this.mute$.next(!currentMuted);
+    }
+
+    private updateConversationIsMuted() {
+        this.mute$.pipe(debounceTime(100), distinctUntilChanged(), takeUntil(this.ngUnsubscribe)).subscribe((isMuted) => {
+            const courseId = this.course()?.id;
+            const conversationId = this.activeConversation()?.id;
+
+            if (!courseId || !conversationId) return;
+
+            this.conversationService.updateIsMuted(courseId, conversationId, isMuted).subscribe({
+                next: () => {
+                    this.activeConversation()!.isMuted = isMuted;
+                    this.onChangePerformed();
+                },
+                error: (errorResponse: HttpErrorResponse) => onError(this.alertService, errorResponse),
+            });
+        });
+    }
+
+    private loadNotificationSettings() {
+        const courseId = this.course()?.id;
+        if (!courseId) {
+            return;
+        }
+
+        this.courseNotificationSettingService.getSettingInfo(courseId).subscribe({
+            next: (response) => {
+                if (response.body) {
+                    this.notificationSettings = response.body;
```

Comment:
I think the github action errors are corect here. you should chcek for undefined and otheer potential risks. maybe sth like this works (next comment)